server/progress_logs/publications.js

```diff
@@ -0,0 +1,29 @@
+Meteor.publish("progressLogsByStudyGroupId", function(limit, id){
+  check(limit, Number);
+  check(id, String);
+
+  let query = new Object();
+  query['study_group.id'] = id;
```

Comment:
Any reason why we're using bracket notation here and dot-notation everywhere else? We should be consistent.

Reply:
Looking forward to your PR fixing all other bracket notations

Reply:
the schema of the **progress log** is as follows.
```
{
  _id : String,
  title : String,
  study_group : [Object]
}

```
So if we want to fetch **progress log** by a study groups our find query would be against the _field_ `study_group.id`.

`query['study_group.id'] = '22'` will result in `{study_group.id: "22"}`

here JS runs toString() on square brackets so we could use special characters into our keys.
with `dot-notation` that does not seem to be feasible.